### PR TITLE
Add readme for TOTP endpoints

### PR DESCRIPTION
**Pull request by hitmanmcc**
This PR adds a new document that describes the endpoints of the authentication methods endpoints.

### CODE REVIEW

**Comment by Americas** [comment]
This could also be an oportonity to update the OTP headers everywhere.

**Comment by hitmanmcc** [comment]
@Americas I think that was outside the scope of this PR in particular but I went ahead and updated it anyway. So you can check that and see it it's ok.

File: _authentication.md [hunk 1]
```diff
@@ -244,4 +244,4 @@ curl https://api.uphold.com/v0/me \
 
 You can use Basic Authentication by providing your username or email and password combination.
 
-If OTP (One Time Password, also known as Two-Factor Authentication) is required, then you will get an HTTP 401 (Unauthorized) response, along with the HTTP header `OTP-Token: Required`. Execute the command above again, this time passing your OTP verification code as a header, like so: `OTP-Token: <OTP-Token>`.
+If OTP (One Time Password, also known as Two-Factor Authentication) is required, then you will get an HTTP 401 (Unauthorized) response, along with the HTTP header `OTP-Token: Required` or `OTP-Method-Id: Required`. For the former case, execute the command above again, this time passing your OTP verification code and phone id as a headers, like so: `OTP-Phone-Id: <OTP-Phone-Id>` and `OTP-Token: <OTP-Token>`. For the latter case, execute the command above again, this time passing your OTP verification code and method id as a headers, like so: `OTP-Method-Id: <OTP-Phone-Id>` and `OTP-Token: <OTP-Token>`.
```
Comment (Americas) [inline _authentication.md]: `OTP-Phone-Id` should be `OTP-Method-Id`?

Comment (hitmanmcc) [reply _authentication.md]: Indeed. 👍

**Comment by hitmanmcc** [comment]
@Americas updated.

File: _authentication.md [hunk 2]
```diff
@@ -244,4 +244,4 @@ curl https://api.uphold.com/v0/me \
 
 You can use Basic Authentication by providing your username or email and password combination.
 
-If OTP (One Time Password, also known as Two-Factor Authentication) is required, then you will get an HTTP 401 (Unauthorized) response, along with the HTTP header `OTP-Token: Required`. Execute the command above again, this time passing your OTP verification code as a header, like so: `OTP-Token: <OTP-Token>`.
+If OTP (One Time Password, also known as Two-Factor Authentication) is required, then you will get an HTTP 401 (Unauthorized) response, along with the HTTP header `OTP-Token: Required` or `OTP-Method-Id: Required`. For the former case, execute the command above again, this time passing your OTP verification code and phone id as a headers, like so: `OTP-Phone-Id: <OTP-Phone-Id>` and `OTP-Token: <OTP-Token>`. For the latter case, execute the command above again, this time passing your OTP verification code and method id as a headers, like so: `OTP-Method-Id: <OTP-Method-Id>` and `OTP-Token: <OTP-Token>`.
```
Comment (waldyrious) [inline _authentication.md]: `One-Time` (with hyphen)

Comment (waldyrious) [reply _authentication.md]: Also, this is personal preference, but I'd start each sentence on its own line (in the code; they'd remain as a single paragraph in the rendered text).

File: _totp.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+### Request
+
+`DELETE https://api.uphold.com/v0/me/authentication_methods/3f8f8264-2f5e-4b2b-8333-473715ab039a`
+
+<aside class="notice">Requires the <code>OTP-Token</code> header to be sent with a valid TOTP token for an existing authentication method. This means that you cannot delete the last verified method of a user.</aside>
+
+### Response
+
+Returns an http status code of 204 in case of success and no JSON body.
```
Comment (waldyrious) [inline _totp.md]: Perhaps wrap "204" in a `<code>` tag? Also, I'd add a comma before the "and no JSON body", just to make the sentence structure more evident.

Comment (waldyrious) [reply _totp.md]: Please also capitalize "HTTP".

**Comment by hitmanmcc** [comment]
@waldyrious updated.

**Comment by waldyrious** [comment]
LGTM, though I'm not qualified to review the correctness of the docs, so I'll leave it to someone else to merge.

File: _totp.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  -H 'OTP-Token: <OTP-Token>'
+```
+
+> The above command does not return a JSON response.
+
+### Request
+
+`DELETE https://api.uphold.com/v0/me/authentication_methods/3f8f8264-2f5e-4b2b-8333-473715ab039a`
+
+<aside class="notice">Requires the <code>OTP-Token</code> header to be sent with a valid TOTP token for an existing authentication method. This means that you cannot delete the last verified method of a user.</aside>
```
Comment (Americas) [inline _totp.md]: the first statement, albeit true, does not correlate with the second statement.

You can use the authentication method you are deleting to verify the request. There is, though, a separate check that does not allow a user to delete its last verified authentication method.

Also, if the authentication method being deleted is not verified, then an `OTP-Token` is not required.

File: _authentication.md [hunk 5]
```diff
@@ -244,4 +244,6 @@ curl https://api.uphold.com/v0/me \
 
 You can use Basic Authentication by providing your username or email and password combination.
 
-If OTP (One Time Password, also known as Two-Factor Authentication) is required, then you will get an HTTP 401 (Unauthorized) response, along with the HTTP header `OTP-Token: Required`. Execute the command above again, this time passing your OTP verification code as a header, like so: `OTP-Token: <OTP-Token>`.
+If OTP (One-Time Password, also known as Two-Factor Authentication) is required, then you will get an HTTP 401 (Unauthorized) response, along with the HTTP header `OTP-Token: Required` or `OTP-Method-Id: Required`.
+For the former case, execute the command above again, this time passing your OTP verification code and phone id as a headers, like so: `OTP-Phone-Id: <OTP-Phone-Id>` and `OTP-Token: <OTP-Token>`.
```
Comment (Americas) [inline _authentication.md]: We should remove mentions of `OTP-Phone-Id`, or at least inform that it is deprecated.

**Comment by hitmanmcc** [comment]
@Americas updated.

File: _totp.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+  "verifiedAt": null
+}
+```
+
+### Request
+
+`POST https://api.uphold.com/v0/me/authentication_methods/totp`
+
+<aside class="notice">Requires the <code>OTP-Method-Id</code> header to be sent with the id of the phone associated with the user's profile information.</aside>
+<aside class="notice">Requires the <code>OTP-Token</code> header to be sent with a valid TOTP token, in cases where there was an already existing authentication method set in place.</aside>
```
Comment (rplopes) [inline _totp.md]: Not sure what `in cases where there was an already existing authentication method set in place` means. WDYT about replacing it with something like `belonging to the authentication method specified in <code>OTP-Method-Id</code>`?

File: _totp.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+  "label": "Authenticator TOTP",
+  "type": "totp",
+  "verified": true,
+  "verifiedAt": "2019-02-11T14:31:48.485Z"
+}
+```
+
+### Request
+
+`POST https://api.uphold.com/v0/me/authentication_methods/3f8f8264-2f5e-4b2b-8333-473715ab039a/verify`
```
Comment (rplopes) [inline _totp.md]: On other pages in our documentation we seem to be using `:id` instead of an actual example id.

File: _authentication.md [hunk 8]
```diff
@@ -244,4 +244,6 @@ curl https://api.uphold.com/v0/me \
 
 You can use Basic Authentication by providing your username or email and password combination.
 
-If OTP (One Time Password, also known as Two-Factor Authentication) is required, then you will get an HTTP 401 (Unauthorized) response, along with the HTTP header `OTP-Token: Required`. Execute the command above again, this time passing your OTP verification code as a header, like so: `OTP-Token: <OTP-Token>`.
+If OTP (One-Time Password, also known as Two-Factor Authentication) is required, then you will get an HTTP 401 (Unauthorized) response, along with the HTTP header `OTP-Token: Required` or `OTP-Method-Id: Required`.
+For the former case, execute the command above again, this time passing your OTP verification code and phone id as a headers, like so: `OTP-Phone-Id: <OTP-Phone-Id>` (deprecated) and `OTP-Token: <OTP-Token>`.
```
Comment (rplopes) [inline _authentication.md]: If this is deprecated, I don't think we should be documenting it here. Let's encourage developers to use the non-deprecated API.

File: _authentication.md [hunk 9]
```diff
@@ -244,4 +244,6 @@ curl https://api.uphold.com/v0/me \
 
 You can use Basic Authentication by providing your username or email and password combination.
 
-If OTP (One Time Password, also known as Two-Factor Authentication) is required, then you will get an HTTP 401 (Unauthorized) response, along with the HTTP header `OTP-Token: Required`. Execute the command above again, this time passing your OTP verification code as a header, like so: `OTP-Token: <OTP-Token>`.
+If OTP (One-Time Password, also known as Two-Factor Authentication) is required, then you will get an HTTP 401 (Unauthorized) response, along with the HTTP header `OTP-Token: Required` or `OTP-Method-Id: Required`.
```
Comment (rplopes) [inline _authentication.md]: We can also receive both headers at once, if no information is provided. Maybe change to:

> along with the HTTP headers `OTP-Token: Required` and/or `OTP-Method-Id: Required`.

File: _authentication.md [hunk 10]
```diff
@@ -244,4 +244,6 @@ curl https://api.uphold.com/v0/me \
...
+For the former case, execute the command above again, this time passing your OTP verification code and phone id as a headers, like so: `OTP-Phone-Id: <OTP-Phone-Id>` (deprecated) and `OTP-Token: <OTP-Token>`.
+For the latter case, execute the command above again, this time passing your OTP verification code and method id as a headers, like so: `OTP-Method-Id: <OTP-Method-Id>` and `OTP-Token: <OTP-Token>`.
```
Comment (rplopes) [inline _authentication.md]: Nit-pick: this starts by referencing token + id, but when showing the headers it's id + token. Keep the same order, for better readability?

File: _totp.md [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+  "type": "authy",
+  "verified": true,
+  "verifiedAt": "2019-01-11T14:20:04.055Z"
+}]
+```
+
+Retrieves a list of authentication methods for the current user.
+
+### Request
+`GET https://api.uphold.com/v0/me/authentication_methods`
```
Comment (rplopes) [inline _totp.md]: We seem to be using newlines between headers and text elsewhere.

File: _authentication.md (hunk 8 above)
Comment (hitmanmcc) [reply _authentication.md]: I disagree, if we support both cases then we have to have both cases documented. I think we should only drop this, when/if we drop support for Authy. New developers coming in to our documentation should know how to be able to deal with authenticating users with accounts still using Authy.

File: _totp.md [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+  "label": "Authenticator TOTP",
+  "type": "totp",
+  "verified": true,
+  "verifiedAt": "2019-02-11T14:31:48.485Z"
+},
+{
+  "default": true,
+  "id": "be95ed5f-d048-4348-9572-411df23bedc9",
+  "label": "+XXXXXXXXXX57",
+  "type": "authy",
```
[the remark this answers is not in the document]
Comment (hitmanmcc) [reply _totp.md]: They are still something that we support.

**Comment by hitmanmcc** [comment]
@rplopes updated.

File: _totp.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (rplopes) [reply _totp.md]: My point was that we didn't need to show an example of something that is optional and deprecated, but fair enough, I see your point that if users still have authy authentication methods, developers should be prepared to handle them 👍

File: _authentication.md (hunk 8 above)
Comment (rplopes) [reply _authentication.md]: We still support authy with `OTP-Method-Id`, so we should encourage developers to use that. `OTP-Phone-Id` is deprecated for developers and they should use `OTP-Method-Id` instead, which replaces it.

Comment (hitmanmcc) [reply _authentication.md]: I was unaware of that fact, in that case, I agree that we can drop the deprecated header from this text.

Comment (hitmanmcc) [reply _authentication.md]: Done.

File: _totp.md [hunk 13]
```diff
@@ -0,0 +1,131 @@
+# Time-Based One-Time password (TOTP)
+
+Uphold provides a TOTP mechanism to secure user accounts. Adopting and adhering to this mechanism is recommended for safety reasons.
+The following section documents how the Authentication Methods API works to provide support for this security mechanism.
+
+## List authentication methods
```
Comment (rplopes) [inline _totp.md]: These should also be title cased (also the ones below).

File: _totp.md [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+Retrieves a list of authentication methods for the current user.
+
+### Request
+
+`GET https://api.uphold.com/v0/me/authentication_methods`
+
+### Response
+Returns an array of the current user's authentication methods.
```
Comment (rplopes) [inline _totp.md]: Still missing newline here.

File: _totp.md [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+### Request
+
+`POST https://api.uphold.com/v0/me/authentication_methods/:id/verify`
+
+
+### Response
+
+Returns an object representing the verified authentication method.
```
Comment (rplopes) [inline _totp.md]: Include the `[Authentication Method](#authentication-method-object)` like with the create endpoint?

**Comment by hitmanmcc** [comment]
@rplopes updated.

**Pull request merged**